.github/workflows/bazel.yml

```diff
@@ -0,0 +1,21 @@
+name: Bazel CI
+on: [push, pull_request, workflow_dispatch]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```

Comment:
Nit: IIUC this is also used for post merge (`on.push` for `main`). With `cancel-in-progress` you are loosing the signal on which commit might have broken `main` as some of the workflows might get cancelled when multiple PRs are merged quickly.

Reply:
Removed, thanks.

Reply:
For pull requests this is usually desired though. Also note that a push to a PR branch will always create two workflow runs (push and pull_request). So I usually create a workflow with on.workflow_call and then separate entrypoints for pull request and push to default branch.